webpush/encrypt.go

```diff
@@ -276,14 +358,18 @@ func hkdf(salt, ikm, info []byte, length int) []byte {
 }
 
 // Encrypt the plaintext message using AES128/GCM
-func encrypt(plaintext, key, nonce []byte) ([]byte, error) {
-	// Add padding. There is a uint16 size followed by that number of bytes of
-	// padding.
-	// TODO: Right now we leave the size at zero. We should add a padding option
-	// that allows the payload size to be obscured.
-	padding := make([]byte, 2)
-	data := append(padding, plaintext...)
-
+func encrypt(plaintext, key, nonce []byte, version Version) ([]byte, error) {
+	var data []byte
+	if version == AES128GCM {
+		data = append(plaintext, []byte{0x02}...)
```

Comment:
"The padding delimiter octet MUST be checked, values other than 0x02 MUST cause the message to be discarded."

From: https://tools.ietf.org/html/draft-ietf-webpush-encryption-08#section-4

Added a comment there as well